src/_pytest/python.py

```diff
@@ -1096,6 +1097,37 @@ def _validate_if_using_arg_names(self, argnames, indirect):
                         pytrace=False,
                     )
 
+    def _validate_explicit_parameters(self, argnames, indirect):
+        """
+        The argnames in *parametrize* should either be declared explicitly via
+        indirect list or explicitly in the function
+
+        :param List[str] argnames: list of argument names passed to ``parametrize()``.
+        :param indirect: same ``indirect`` parameter of ``parametrize()``.
+        :raise ValueError: if validation fails
+        """
+        func_name = self.function.__name__
+        if type(indirect) is bool and indirect is True:
+            return
+        temp_argnames = list()
+        if type(indirect) is list:
+            for arg in argnames:
+                if arg not in indirect:
+                    temp_argnames.append(arg)
+        elif indirect is False:
+            temp_argnames = argnames
+        for arg in temp_argnames:
+            # reference: https://docs.python.org/3/library/inspect.html#inspect.getfullargspec
+            # this will contain args, vargargs, and everything python function can have
+            fullargspec = inspect.getfullargspec(self.function)
```

Comment:
Use `_pytest.compat.getfuncargnames` instead here as it handles some special cases already. Also move this call out of the loop.

Reply:
Done!

Reply:
Done!